### DISCUSSION TITLE
Updating UT to work with CMake build system

### DISCUSSION THREAD

**Discussion by shuq0** [nasa/cFS#489] [Q&A] [answered]
I am looking for a methodology or guide to build and run or convert unit tests which are based on gcov for apps which have not been updated such as:
* https://github.com/nasa/CFS_CI
* https://github.com/nasa/CFS_TO
etc.
 The only app updated is: 
 * https://github.com/nasa/CF 

I have viewed the diffs between 2a292d041a3644844517cc10344abbf0c4ecccc2 and 7a48a8b2e8f4f5b103748685f076d9dc8b3b66fb but it is still not clear why things were shifted around. 

Thanks in advance.

**Comment by skliper** [comment, marked as answer]
CF, CS, DS, FM, HK, HS, LC, MM, MD, SC as well as the framework (cFE/osal/psp) have all been updated to use https://github.com/nasa/osal/tree/main/ut_assert as of the latest app releases.  The "old" UT was problematic due to the stubs being owned by UT which complicated maintenance since the repos own the APIs.  Shifting to osal/ut_assert where the stubs are maintained within each repo (cFE owns the cFE stubs, osal owns the osal stubs, etc) allows for testing with a user defined set of repositories/versions (vs API versions being "fixed" from UT).  The recommendation is to update cfs_ci and cfs_to to use osal/ut_assert vs updating the old UT to work with the modern framework.
